src/recipient.js

```diff
@@ -55,6 +56,24 @@ function Recipient(name) {
   // years of earning history: total earnings / 35 / 12.
   this.monthlyIndexedEarnings = 0;
 
+  // The total credits earned, maximum of 40
+  this.earnedCredits = 0;
+
+  // The total of credits that would be earned in the planned work
+  this.plannedCredits = 0;
+
+  // total credits = earnedCredits + planned credits
+  this.totalCredits = 0;
+
+  // Have earnings before 1978
+  this.earningsBefore1978 = false;
```

Comment:
nit, totally optional suggestion: I like to name boolean variables using a verb phrase that indicates it's a boolean. So hasEarningsBefore1978. Especially since javascript doesn't have a types, it's easy to make a mistake elsewhere in the code and think this is a number such as:

```
if (this.earningsBefore1978 > 250)
```
or something, where:

```
if (this.hasEarningsBefore1978 > 250)
```

is more obviously incorrect. You didn't actually make any mistakes or anything, just something that helps me to prevent them.

Reply:
Fixed.